vertx-sql-client/src/main/java/io/vertx/sqlclient/spi/Driver.java

```diff
@@ -27,6 +35,71 @@
  */
 public interface Driver {
   
+  public static enum KnownDrivers {
```

Comment:
this creates an implicit dependency (yet weak) from sql-client onto the existing implementations. We are actually going to have a JDBC implementation as well that will not be in this GitHub repository,  so I think it would be best better to avoid this and have instead each driver define a string constant for this.

Reply:
When we add a JDBC implementation in a separate repo we can add another value to this enum, right?
Any other drivers that come along will still work, and will ultimately be able to define what they want their string constants to be.
Mainly I'm just going for user-convenience here by adding "knowledge of" the known implementations, I don't know if you could really call it a dependency

Reply:
there isn't much a point of having `KnownDrivers` enum than having a glorified factory, i.e

if you write

```java
SqlConnectOptions options = Driver.createConnectOptions(Driver.KnownDrivers.POSTGRESQL)
```

then you can simply write

```java
SqlConnectOptions options = new PgConnectOptions();
```

Reply:
I was mainly looking at this from a UX perspective, but since this is just the SPI path (not API) UX is less important because users won't be doing this directly.
I'll go ahead and remove the `KnownDrivers` enum and associated code